content/ADR-145-pick-realm-v2.md

```diff
@@ -9,43 +9,63 @@ spdx-license: CC0-1.0
 
-The current implementation (as described in [Realm Picking Algorithm ADR](/adr/ADR-86)) is expansible by adding new links. In this proposal a filter and a new link will be added: 
-- Accepting Users Filter: Will filter out if the amount of resources for that Catalyst is at limit.
-- Version Link: the algorithm will prioritize sending traffic to instances that have the latest version
+### Algorithm improvement
 
+The existing implementation of the realm picking algorithm, as outlined in the [Realm Picking Algorithm ADR](https://adr.decentraland.org/adr/ADR-86), allows for extensibility through the addition of new links and filters. In this proposal, two filters and a new link will be introduced:
+- **OVERLOADED_CATALYST**: This filter will exclude a Catalyst if its resource usage or user capacity reaches the limit.
+- **VERSION_CATALYST**: This filter will exclude a Catalyst if it does not meet the minimum required version as specified.
+- **FORCE_CATALYST**: This new link will enable the forcible selection of a Catalyst listed in its configuration.
+
+Furthermore, four new variants will be added, each containing a set of rules designed to address specific contexts.
+
+### Server-side
+
+This algorithm will be implemented on the server-side to accommodate various client implementations. While this may require modifying the implementation of the `LARGE_LATENCY` rule to receive necessary information from the client-side, it will enable enhancements such as adding a cache mechanism or event-driven notifications to proactively filter out unhealthy Catalysts, thereby improving the overall performance of the algorithm.
+
+To fulfill the goal of creating a cost-effective, scalable, and globally accessible service, two implementation proposals, CloudFlare Workers and AWS Lambdas, are being considered. The optimal choice for this service will be determined by analyzing various aspects related to cost, pricing model, performance, and latency, which will be discussed in detail in the Server-Side Implementation section.
+
+To assess these aspects, we will examine the workload of the current algorithm, which is monitored through Segment events. The chart below presents the workload data for the past 4 months as of the time of writing.
+
+<img src="resources/ADR-145/realm-picking-algorithm-workload.png" alt="drawing" style="width:100%;"/>
+
+_The average workload is 10k._
 
 ## Specification
 
-### Accepting Users Filter
+### Overloaded Catalyst
 
 _Description_:
 
-Publicly exposing the Catalyst Node's available resources is a security risk, so their estimation should be done in the Catalyst (specifically in the BFF) and not in the Kernel.
-
-This way, the BFF will leverage the usage of the field `accepting_users` in the `/about` endpoint.
+Exposing the available resources of Catalyst Nodes publicly poses a security risk, and thus, their estimation should be conducted within the Catalyst itself (_specifically in the BFF_), rather than in the Kernel.
```

Comment:
Should we still mention BFF here?

Reply:
Currently this is done in the BFF, may it is worth keeping it like this for now and update the ADR once we completely remove the BFF, what do you think?

Reply:
If the BFF is about to be removed, then we shouldn't mention it here. Anyways, we should specify here who will be responsable for this

Reply:
I won't mention the BFF, we can mention the realm description without specifying who servers, specially since the endpoint is `/about` not `/bff/about`, so bff is just an implementation detail here

Reply:
Got it, I'm removing this mention about the BFF. Is it enough to mention that these versions are going to be retrieved from each Catalyst?